subprojects/core/src/integTest/groovy/org/gradle/api/file/FileCollectionSymlinkIntegrationTest.groovy

```diff
@@ -65,6 +66,74 @@ class FileCollectionSymlinkIntegrationTest extends AbstractIntegrationSpec {
         "project.objects.fileCollection()"   | "project.objects.fileCollection().from(file, symlink, symlinked)"
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/1365")
+    def "detect changes to broken symlink outputs"() {
+        def root = file("root").createDir()
+        def target = file("target")
+        def link = root.file("link")
+
+        buildFile << """
+            import java.nio.file.*
+            class ProducesLink extends DefaultTask {
+                @OutputDirectory File outputDirectory // TODO outputfile?
+    
+                @TaskAction execute() {
+                    def link = Paths.get('${link}')
```

Comment:
```suggestion
                    def link = outputDirectory.toPath().resolve('${link.name}'))
```

Reply:
I'm not yet sure about why absolute paths aren't preferred, please elaborate